### ISSUE TITLE
Remirror cannot be used in ESM node project

### ISSUE THREAD

**Issue by zofiag** [remirror/remirror#1604]
### Summary

NodeJS has supported ES modules since version 12 but `remirror` doesn't seem to work for Node with ESM enabled.

### Steps to reproduce

1. Create a project where `package.json` has `"type": "module"` (alternatively you can use `.mjs` extension for `.js` files), so that ESM are enabled
2. Install one of `remirror` packages eg `@remirror/pm`
3. Create `index.js` (or `index.mjs` if you don't have `"type": "module"` set in your `package.json`)
4. Add a following code to the `index.js` file
```javascript
import { Node } from "@remirror/pm/model";
console.log(Node);
```
5. Run `node index.js` (or `node index.mjs`)

[reproduction repository](https://github.com/zofiag/esm/tree/remirror)

### Expected results

Code runs without any issue and prompts:`[Function: Node] { fromJSON: [Function: fromJSON] }`.

### Actual results

An error is thrown
```
import { Node } from "@remirror/pm/model";
         ^^^^
SyntaxError: Named export 'Node' not found. The requested module '@remirror/pm/model' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@remirror/pm/model';
const { Node } = pkg;

    at ModuleJob._instantiate (node:internal/modules/esm/module_job:127:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:191:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:337:24)
    at async loadESM (node:internal/process/esm_loader:88:5)
    at async handleMainPromise (node:internal/modules/run_main:61:12)
```

### Possible Solution

Point all `import` entries under `exports` in all `package.json` files to `.mjs` extension instead of `.js`. For example:
```json
// node_modules/@remirror/pm/package.json
{
  "name": "@remirror/pm",
  "exports": {
      "./model": {
        "import": "./model/dist/remirror-pm-model.esm.mjs",
      }
   }
}
```
NodeJS module resolution treats all of the `.js` files as CommonJS and all `.mjs` files as ES modules.

**Comment by ocavue** [comment]
I agree we should rename `.esm.js` to `.esm.mjs`. Node doesn't allow having files with the same extension to be treated as `cjs` or `esm` at the same time. ([Detailed explanation](https://github.com/sheremet-va/dual-packaging)). We didn't see this issue before because of we was using remirror under bundlers (webpack etc), which don't have this limitation.

**Comment by zofiag** [comment]
I wonder if there's anything that can be done now or is it blocked because of this preconstruct open issue https://github.com/preconstruct/preconstruct/issues/251?

**Comment by ocavue** [comment]
Since preconstruct doesn't support it, we will need to get rid of preconstruct or use some ugly hack. In any case, we certainly won't be able to fix this in a very short term.

**Comment by whawker** [comment]
@zofiag this could end up happening sooner than we thought. 
As ProseMirror is releasing it's own typescript definitions we likely need to release a new major version due to breaking type changes. Therefore dropping CommonJS in favour of ESM is being considered for the major

**Comment by ocavue** [comment]
This is fixed in the beta version.

```
bash-3.2$ pnpm add "@remirror/pm@beta"
dependencies:
+ @remirror/pm 2.0.0-beta.1
Progress: resolved 42, reused 40, downloaded 2, added 42, done
bash-3.2$
bash-3.2$ echo 'import { Node } from "@remirror/pm/model"; console.log(Node);' > file.mjs
bash-3.2$
bash-3.2$ node file.mjs
[class Node]
bash-3.2$
```
